src/Scripting/ScriptingSystem.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <Library/Logger/LogCategory.h>
+
+#include <string_view>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <memory>
+#include <utility>
+#include <sol/sol.hpp>
+
+class DistLogSink;
+class IBindings;
+class PlatformApplication;
+class ScriptLogSink;
+
+class ScriptingSystem {
+ public:
+    ScriptingSystem(std::string_view scriptFolder, std::string_view entryPointFile, PlatformApplication &platformApplication, DistLogSink &distLogSink);
+    ~ScriptingSystem();
+
+    void executeEntryPoint();
+
+    template<typename TBindings, typename ...TArgs>
+    void addBindings(std::string_view bindingTableName, TArgs &&... args) {
+        auto bindings = std::make_unique<TBindings>(std::forward<TArgs>(args) ...);
+        _addBindings(bindingTableName, std::move(bindings));
+    }
+
+    static LogCategory LogCategory;
+
+ private:
+    void _initBaseLibraries();
+    void _initPackageTable(std::string_view scriptFolder);
+    void _addBindings(std::string_view name, std::unique_ptr<IBindings> bindings);
+
+    std::shared_ptr<sol::state> _solState;
```

Comment:
I'm going to use a `shared_ptr` mostly to avoid cases where the `sol::state` is being destroyed but its `sol::state_view` are still alive somewhere.
`sol2` doesn't have a way to tell the state_view that the lua state is being closed. ( The main goal of a `state_view` is to be as lightweight as possible ).

So I'm passing a std::weak_ptr to those classes that lives outside of the `ScriptingSystem` and requires a view. In this case we have `ScriptLogSink` and `InputScriptEventHandler`.